models/Employee.js

```diff
@@ -7,39 +7,70 @@ const employeeSchema = new Schema({
   email: { type: String, required: true, unique: true },
   role: { type: String, default: 'Employee' },
 
+  onboardingStatus: {
+    type: String,
+    enum: ['Not Started', 'Pending', 'Approved', 'Rejected'],
+    default: 'Not Started',
+  },
+
+  // profilePicture: { type: String, default: 'path/to/default/profile-picture.png' },
+  profilePicture: { type: String},
+
   // Onboarding and Profile Information (optional for now)
-  firstName: String,
-  lastName: String,
-  middleName: String,
+  firstName: { type: String, required: true },
+  lastName: { type: String, required: true },
+  middleName: { type: String },
+  preferredName: { type: String },
+
   address: {
     street: String,
     city: String,
     state: String,
     zip: String,
+    required:true
   },
-  phoneNumber: String,
-  workPhoneNumber: String,
-  ssn: String,
-  dob: Date,
-  gender: String,
-  workAuthorization: {
-    type: String, // e.g., "H1-B", "OPT", etc.
-    startDate: Date,
-    endDate: Date,
-    documents: [String], // list of document URLs or references
+  cellPhoneNumber: { type: String, required: true},
+  workPhoneNumber: { type: String },
+  car: {
+    make: { type: String },
+    model: { type: String },
+    color: { type: String }
+  },
+  ssn: { type: String, required: true },
+  dateOfBirth: { type: Date, required: true },
+  gender: { type: String, enum: ['Male', 'Female', 'I do not wish to answer'] },
+  citizenship: {
```

Comment:
Hi Kha, I think we might not have to store boolean logic here. We can do this logic in UI. There can be options for the user to click, and we can pass that click value to the server and store it in the database. For the status 'other', we can allow the user to type in the value and store it in the database.
The schema I expected is:

visaStatus: {type: String},   // to store 'Green Card', 'Citizen', 'H1-B', 'L2', 'F1(CPT/OPT)', 'H4', 'Other
document: { visa: { type: String},    // use this name to find the file in S3 bucket
                    startDate: { type: Date },
                    endDate: { type: Date } },
optDocument: { type: mongoose.Schema.Types.ObjectId, ref:'VisaDocuments'}   // store documents object Id here, which is  _id in VisaDocuments schema

To make the schema simple, what do you think?